docs/src/filesystem-sysroot.md

```diff
@@ -0,0 +1,69 @@
+# Filesystem: Physical /sysroot
+
+The bootc project uses [ostree](https://github.com/ostreedev/ostree/) as a backend,
+and maps fetched container images to a "deployment": <https://ostreedev.github.io/ostree/deployment/>.
+
+## stateroot
+
+The underlying `ostree` CLI and API tooling expose a concept of `stateroot`, which
+is not yet exposed via `bootc`.  The `stateroot` used by `bootc install`
+is just named `default`.
+
+The stateroot concept allows having fully separate parallel operating
+system installations with fully separate `/etc` and `/var`, while
+still sharing an underlying root filesystem.
+
+In the future, this functionality will be exposed and used by `bootc`.
+
+## /sysroot mount
+
+When booted, the physical root will be available at `/sysroot` as a
+read-only mount point.  This is a key aspect of how `bootc upgrade` operates:
+it fetches the updated container image and writes new files to `/sysroot/ostree`.
+
+Beyond that and debugging/introspection, there are few use cases for tooling to
+operate on the physical root.
+
+## Expanding the root filesystem
+
+One notable use case that *does* need to operate on `/sysroot`
+is expanding the root filesystem.
+
+Some higher level tools such as e.g. `cloud-init` may (reasonably)
+expect the `/` mount point to be the physical root.  Tools like
+this will need to be adjusted to instead detect this and operate
+on `/sysroot`.
+
+### Growing the block device
+
+Fundamentally bootc is agnostic to the underlying block device setup.
+How to grow the root block device depends on the underlying
+storage stack, from basic partitions to LVM.  However, a
+common tool is the https://manpages.debian.org/testing/cloud-guest-utils/growpart.1.en.html[growpart]
```

Comment:
bad link, should be - `[growpart](https://manpages.debian.org/testing/cloud-guest-utils/growpart.1.en.html)`

Reply:
Ah yes...hmm, I need to figure out if there's some markdown linter which can find this stuff.  The asciidoc vs markdown thing...argh